### ISSUE TITLE
Chain ignores blocks from peers if we are only one behind.

### ISSUE THREAD

**Issue by ggoranov** [dusk-network/dusk-blockchain#468]
As per the code snippet below, Chain ignores blocks from peers if we are only one behind. Kadcast does not work with Synchronizer. As a result, topics.Block message that came from Kadcast are always only-one-behind, respectively rejected by Chain.

```golang

func (c *Chain) onAcceptBlock(m message.Message) error {
	// Ignore blocks from peers if we are only one behind - we are most
	// likely just about to finalize consensus.
	// TODO: we should probably just accept it if consensus was not
	// started yet
	if !c.counter.IsSyncing() {
		return nil
	}

```

**Comment by julesdesmit** [comment]
I feel like the cleanest way to solve this would be to refactor the `Synchronizer` a bit. Instead of having a `Synchronizer` on each single peer, sharing a `Counter`, there could be one single `Synchronizer` that all recently received blocks are pushed to. That `Synchronizer` could then just serially process the incoming blocks, allowing kadcast peers to send their blocks here as well.

**Comment by autholykos** [comment]
This is a problem with topic propagation and needs to be handled before switching to Kadcast

**Comment by autholykos** [comment]
Closed via #707
